main.rkt

```diff
@@ -285,7 +289,63 @@
              #:with (c:strict-test) (syntax/loc #'a (#:t a))
              #:attr unit #'c.unit]))
 
-(define-simple-macro (chk  e:test ... ) (let () e.unit ...))
+;; this should match any string which is a valid racket identifier
+;; and does not contain the '=' character.
+(define id-regexp "[^]\\[\\(\\){}\",\'`;#\\|\\\\=]+")
+(define key-val-regexp (regexp (string-append id-regexp "=.+")))
+
+(define (get-filters)
+  (for/fold ([names empty] [filters empty] [files empty] [lines empty])
+            ([arg (vector->list (current-command-line-arguments))])
+    (cond
+      [(regexp-match-exact? #rx"[Ff][Ii][Ll][Ee]=.+" arg)
+       (define new-files (cons (regexp (substring arg 5)) files))
+       (values names filters new-files lines)]
+      [(regexp-match-exact? #rx"[Ll][Ii][Nn][Ee]=.+" arg)
+       (define new-lines (cons (string->number (substring arg 5)) lines))
+       (values names filters files new-lines)]
+      [(regexp-match-exact? key-val-regexp arg)
+       (define split (string-split arg "="))
+       (define arg-name (string->symbol (car split)))
+       (define arg-val (cadr split))
```

Comment:
I changed it so that `arg-val` is the result of re-joining `(rest split)`, so a key/val pair such as `foo=abc=def` will be parsed into `('foo . "abc=def")`. Previously, it would ignore the `=def` and parse it into `('foo . "abc")`